### DISCUSSION TITLE
Automatically moving the goti, if only one goti of that colour is there

### DISCUSSION THREAD

**Discussion by adi-g15** [adi-g15/Ludo-The_Game#40] [General]
See for example:

Isme jab blue ka turn hoga, tab say dice gives 1-5, to hume chose karna hota hai, but actually ek hi option hai.

![image](https://user-images.githubusercontent.com/37269665/106321834-e025ee80-629a-11eb-8824-a0611ce07cf8.png)

What if, we automatically move the goti ?

**Comment by adi-g15** [comment]
@Athi223 What are your thoughts on this ? Since you best know the code, is it trivial to implement this, or there maybe any big changes ?
Ek problem bhi hai isme, agar automatic move karta hai, to ye `instant` feel ho sakta hai, turn aaya aur gaya 😅, that may not be what you wanted.

**Reply by Athi223** [reply]
> @Athi223 What are your thoughts on this ? Since you best know the code, is it trivial to implement this, or there maybe any big changes ?

Its probably a trivial task, since we're storing the state(and position if applicable) of every _goti_ in the `positions` array, all we have to do is check if the array contains only `1 string-element`(since locked and completed gotis are stored as `false` and `true` respectively) and if so, call the `move` function directly instead of prompting the user to click a _goti_. All this of course assuming there's no other hinderence to these steps.

> Ek problem bhi hai isme, agar automatic move karta hai, to ye `instant` feel ho sakta hai, turn aaya aur gaya 😅, that may not be what you wanted.
For the automatic part, I guess all we could do is add a delay of 0.5/1 sec, but I'm not sure how safe that is for the overall working of the app(if incase it causes any delay to our API calls or any such thing) & also for User Experience.

TLDR; Move animation would probably be a tedious task.
Kyuki dusra tareeka hoga goti ko animate karna (in all cases) but that probably sounds like a tedious task of getting coordinates of source & destination and stuff like that, also goti size change karni pad sakti hai since we reduce size when multiple gotis stay in same block.

**Reply by Athi223** [reply]
I'll try to work on this in a couple of days since abhi mera project report cha thoda kam chal raha tha, if you're okay with it.
Else if you want to give it a try yourself, the steps would roughly be as I mentioned above.

**Reply by adi-g15** [reply]
I will give it a try 👍.
You don't actually have obligation to separately focus on this too, there's no hurry 😁👍.

**Reply by Athi223** [reply]
bro is it done already?

**Reply by adi-g15** [reply]
Nahi, after knowing a bit of CSS animation, it may not have been much good than current behaviour I left it as is. 🙄

**Reply by Athi223** [reply]
haha yeah i had mentioned that already, also kaafi tedious hoga wo.
I was asking about the _automatic move_ vala part. Also I'm not totally sure but shayad vo part tha already mere code mein.

**Reply by adi-g15** [reply]
Automatic move maine add nahi kiya tha, kyunki phir wahi problem hota ki, it moved within a second, and waiting for next move 😅
